include/bmi/AbstractCLibBmiAdapter.hpp

```diff
@@ -27,31 +26,17 @@ namespace models {
              */
             AbstractCLibBmiAdapter(const std::string &type_name, std::string library_file_path, std::string bmi_init_config,
                                    std::string forcing_file_path, bool allow_exceed_end, bool has_fixed_time_step,
-                                   std::string registration_func, utils::StreamHandler output)
-                    : Bmi_Adapter(type_name, std::move(bmi_init_config), std::move(forcing_file_path),
-                                     allow_exceed_end, has_fixed_time_step, output),
-                      bmi_lib_file(std::move(library_file_path)),
-                      bmi_registration_function(std::move(registration_func)) { }
-
-            AbstractCLibBmiAdapter(AbstractCLibBmiAdapter &&adapter) noexcept :
-                    Bmi_Adapter(std::move(adapter)),
-                    bmi_lib_file(std::move(adapter.bmi_lib_file)),
-                    bmi_registration_function(adapter.bmi_registration_function),
-                    dyn_lib_handle(adapter.dyn_lib_handle)
-            {
-                // Have to make sure to do this after "moving" so the original does not close the dynamically loaded library handle
-                adapter.dyn_lib_handle = nullptr;
-            }
+                                   std::string registration_func, utils::StreamHandler output);
+
+            AbstractCLibBmiAdapter(AbstractCLibBmiAdapter &&adapter) noexcept = default;
```

Comment:
The body of the destructor means that the previous explicit body of the move constructor was necessary. A moved-from instance should not be calling `dlclose` on its `dyn_lib_handle`, since that would result in a doubled call.

Rather than restoring/fixing it, does the code build with this
```suggestion
            AbstractCLibBmiAdapter(AbstractCLibBmiAdapter &&adapter)= delete;
```
If so, I think that's preferable for making it enforceably clear that `AbstractCLibBmiAdapter` and its descendants are not value types, and should never be treated as such - consumers should always be constructing in place and assigning pointers, and doing otherwise is likely a mistake.

Reply:
I'll give that a quick test and report back.

Reply:
Ah, yeah I messed this one up. I don't remember why I made this change (I assume I thought that the original handle would be set to nullptr when moved, but that shouldn't work) 🤦🏾 thanks for catching that.

Reply:
At the very least, unit tests for C and Fortran pass when the move constructor is deleted.

Reply:
I've got everything passing locally. Hopefully CI agrees.